test/PaginationSpec.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import ReactTestUtils from 'react/lib/ReactTestUtils';
+import Pagination from '../src/Pagination';
+
+describe('Pagination', function () {
+  it('Should have class', function () {
+    let instance = ReactTestUtils.renderIntoDocument(
+      <Pagination>Item content</Pagination>
+    );
+    assert.ok(ReactTestUtils.findRenderedDOMComponentWithClass(instance, 'pagination'));
+  });
+
+  it('Should show the correct active button', function () {
+    let instance = ReactTestUtils.renderIntoDocument(
+      <Pagination
+        items={5}
+        activePage={3} />
+    );
+    let pageButtons = ReactTestUtils.scryRenderedDOMComponentsWithTag(instance, 'li');
+    assert.equal(pageButtons.length, 5);
+    assert.equal(pageButtons[2].getDOMNode().getAttribute('class').match(/active/)[0], 'active');
+  });
+
+  it('Should call onSelect when page button is selected', function (done) {
+    function onSelect(pagenumber) {
+      assert.equal(pagenumber, 2);
+      done();
+    }
+
+    let instance = ReactTestUtils.renderIntoDocument(
+      <Pagination items={5} onSelect={onSelect} />
+    );
+
+    ReactTestUtils.Simulate.click(
+      ReactTestUtils.scryRenderedDOMComponentsWithTag(instance, 'a')[1]
+    );
+  });
+
+  it('Should only show part of buttons when given maxButtons', function () {
+    let instance = ReactTestUtils.renderIntoDocument(
+      <Pagination
+        items={20}
+        activePage={8}
+        maxButtons={5} />
+    );
+    let pageButtons = ReactTestUtils.scryRenderedDOMComponentsWithTag(instance, 'li');
+    // include ellipsis button
+    assert.equal(pageButtons.length, 6);
+
+    // active button is the second one
+    assert.equal(pageButtons[0].getDOMNode().firstChild.innerText, '7');
+    assert.equal(pageButtons[1].getDOMNode().getAttribute('class').match(/active/)[0], 'active');
+  });
+
+  it('Should show the first and last button', function () {
+    let instance = ReactTestUtils.renderIntoDocument(
+      <Pagination
+        first={true}
+        last={true}
+        items={5} />
+    );
+    let pageButtons = ReactTestUtils.scryRenderedDOMComponentsWithTag(instance, 'li');
+    // add 2 more button
+    assert.equal(pageButtons.length, 7);
+  });
```

Comment:
This test does not assert that the added two buttons are truly the first and last buttons. It would also pass if they were the next and previous buttons.